### PR TITLE
Pre-processing improvements

### PR DESCRIPTION
**Pull request by ssd04**
- cleanup old code
- move events logic from facade to eventsHandler
- return early if duplicated event

### CODE REVIEW

File: process/eventsHandler.go [hunk 1]
```diff
@@ -67,42 +68,79 @@ func checkArgs(args ArgsEventsHandler) error {
...
+	if err != nil {
+		return err
 	}
 
-	if !shouldProcessEvents {
-		log.Info("received duplicated events", "event", common.PushLogsAndEvents,
-			"block hash", events.Hash,
+	pushEvents := data.BlockEvents{
+		Hash:      eventsData.Hash,
+		ShardID:   eventsData.Header.GetShardID(),
```
Comment (sstanculeanu) [inline process/eventsHandler.go:93]: is there any chance here to have the header nil, resulting in a panic?

File: process/eventsHandler.go [hunk 2]
```diff
@@ -252,29 +280,16 @@ func (eh *eventsHandler) HandleBlockScrs(blockScrs data.BlockScrs) {
 }
 
 // HandleBlockEventsWithOrder will handle full block events received from observer
-func (eh *eventsHandler) HandleBlockEventsWithOrder(blockTxs data.BlockEventsWithOrder) {
+func (eh *eventsHandler) handleBlockEventsWithOrder(blockTxs data.BlockEventsWithOrder) {
```
Comment (sstanculeanu) [inline process/eventsHandler.go:283]: update comment with lower case

File: integrationTests/websocket/testNotifierWithWebsockets_test.go [hunk 3]
```diff
@@ -715,6 +715,6 @@ func testNotifierWithWebsockets_AllEvents(t *testing.T, observerType string) {
 
 	integrationTests.WaitTimeout(t, wg, time.Second*4)
 
-	assert.Equal(t, numEvents, len(notifier.RedisClient.GetEntries()))
-	assert.Equal(t, numEvents, len(notifier.RedisClient.GetEntries()))
+	expectedNumRedisEvents := 3 // one redis events per push, revert, finalized
```
Comment (sstanculeanu) [inline integrationTests/websocket/testNotifierWithWebsockets_test.go:718]:
```suggestion
	expectedNumRedisEvents := 3 // one redis event per push, revert, finalized
```

File: api/groups/interface.go [hunk 4]
```diff
@@ -10,8 +10,7 @@ import (
 
 // EventsFacadeHandler defines the behavior of a facade handler needed for events group
 type EventsFacadeHandler interface {
-	HandlePushEventsV2(events data.ArgsSaveBlockData) error
-	HandlePushEventsV1(events data.SaveBlockData) error
+	HandlePushEvents(events data.ArgsSaveBlockData) error
```
Comment (iulianpascalau) [inline api/groups/interface.go:13]: :+1:

File: process/eventsHandler.go (hunk 1 above)
Comment (ssd04) [reply process/eventsHandler.go:93]: no, there is a header check in events interceptor

File: process/eventsHandler.go (hunk 2 above)
Comment (ssd04) [reply process/eventsHandler.go:283]: updated comment

File: integrationTests/websocket/testNotifierWithWebsockets_test.go (hunk 3 above)
Comment (ssd04) [reply integrationTests/websocket/testNotifierWithWebsockets_test.go]: :+1:

**Review by sstanculeanu — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**